startup/25-QEPro.py

```diff
@@ -0,0 +1,550 @@
+                infuse_rate = meta['start']['infuse_rate']
+            else: infuse_rate = ['None']
+            if 'infuse_rate_unit' in meta['start'].keys():
+                infuse_rate_unit = meta['start']['infuse_rate_unit']
+            else: infuse_rate_unit = ['None']
+            if 'pump_status' in meta['start'].keys():
+                pump_status = meta['start']['pump_status']
+            else: pump_status = ['None']
+            if 'mixer' in meta['start'].keys():
+                mixer = meta['start']['mixer']
+            else: mixer = ['None']
+            if 'sample_type' in meta['start'].keys():
+                sample_type = meta['start']['sample_type']
+            # else: sample_type = None
+
+
+        if plot == True:
+            x_axis_label = self.x_axis_format.get(as_string=True)
+            if spectrum_type == 3:
+                y_axis_label = 'Absorbance'
+            elif spectrum_type == 2:
+                y_axis_label = 'Fluorescence'
+            plt.figure()
+            plt.plot(x_axis_data, output_data)
+            plt.xlabel(x_axis_label)
+            plt.ylabel(y_axis_label)
+            plt.show()
+
+        if csv_path != None:
+
+            if spectrum_type == 3:
+                spec = 'Abs'
+                fout = f'{csv_path}/{sample_type}_{spec}_{date}-{time}_{uid[0:8]}.csv'
+
+            if spectrum_type == 2:
+                spec = 'PL'
+                fout = f'{csv_path}/{sample_type}_{spec}_{date}-{time}_{uid[0:8]}.csv'
+
+            with open(fout, 'w') as fp:
+                fp.write(f'uid,{full_uid}\n')
+                fp.write(f'Time_QEPro,{date},{time}\n')
+                fp.write(f'Integration time (ms),{int_time}\n')
+                fp.write(f'Number of averaged spectra,{num_average}\n')
+                fp.write(f'Boxcar width,{boxcar_width}\n')
+
+                for i in range(len(pump_names)):
+                    fp.write(f'{pump_names[i]},{precursor[i]},{infuse_rate[i]},{infuse_rate_unit[i]},{pump_status[i]}\n')
+
+                if mixer != None:
+                    for i in range(len(mixer)):
+                        fp.write(f'Mixer no. {i+1},{mixer[i]}\n')
+
+                if spectrum_type == 3:
+                    fp.write('Wavelength,Dark,Reference,Sample,Absorbance\n')
+                else:
+                    fp.write('Wavelength,Dark,Sample,Fluorescence\n')
+
+                for i in range(len(output_data)):
+                    if spectrum_type == 3:
+                        fp.write(f'{x_axis_data[i]},{dark_data[i]},{reference_data[i]},{sample_data[i]},{output_data[i]}\n')
+                    else:
+                        fp.write(f'{x_axis_data[i]},{dark_data[i]},{sample_data[i]},{output_data[i]}\n')
+
+try:
+    from tiled.client import from_profile
+    tiled_client = from_profile("xpd-ldrd20-31")
+except (NameError, ModuleNotFoundError):
+    pass
+qepro = QEPro('XF:28ID2-ES{QEPro:Spec-1}:', name='QEPro', )
```

Comment:
TODO: make instantiation conditional based on the fact is the equipment is at the beamline or not. `try..except` will work.